diff --git a/resources/testharness.js b/resources/testharness.js
index c5c375e17..a4610d927 100644
--- a/resources/testharness.js
+++ b/resources/testharness.js
@@ -2589,10 +2589,6 @@
      */
     Test.prototype.step = function(func, this_obj)
     {
-        if (this.phase > this.phases.STARTED) {
-            return;
-        }
-
         if (settings.debug && this.phase !== this.phases.STARTED) {
             console.log("TEST START", this.name);
         }
@@ -2621,9 +2617,6 @@
         try {
             return func.apply(this_obj, Array.prototype.slice.call(arguments, 2));
         } catch (e) {
-            if (this.phase >= this.phases.HAS_RESULT) {
-                return;
-            }
             var status = e instanceof OptionalFeatureUnsupportedError ? this.PRECONDITION_FAILED : this.FAIL;
             var message = String((typeof e === "object" && e !== null) ? e.message : e);
             var stack = e.stack ? e.stack : null;
@@ -2631,6 +2624,8 @@
             this.set_status(status, message, stack);
             this.phase = this.phases.HAS_RESULT;
             this.done();
+
+            throw `${this.name} failed - ${e}`;
         } finally {
             this.current_test = null;
         }
