diff --git a/WebCryptoAPI/getRandomValues.any.js b/WebCryptoAPI/getRandomValues.any.js
index 574134eb7..128c805a5 100644
--- a/WebCryptoAPI/getRandomValues.any.js
+++ b/WebCryptoAPI/getRandomValues.any.js
@@ -1,14 +1,31 @@
+// k6 throws GoError instead of DOMException, so we need to monkey patch.
+const originalAssertThrowsDom = self.assert_throws_dom;
+self.assert_throws_dom = function(expectedErrorName, func, description) {
+    try {
+        func();
+    } catch (e) {
+        const isGoError = (typeof e === "object") && (e.name === "GoError");
+        const msg = isGoError ? (e.message || "") : e.toString();
+        if (msg.includes(expectedErrorName)) {
+            return; // Treat as correct
+        }
+    }
+
+    return originalAssertThrowsDom(expectedErrorName, func, description);
+}
+
 // Step 1.
-test(function() {
-    assert_throws_dom("TypeMismatchError", function() {
-        self.crypto.getRandomValues(new Float16Array(6))
-    }, "Float16Array")
+// Float16Array is not supported in k6
+// test(function() {
+//     assert_throws_dom("TypeMismatchError", function() {
+//         self.crypto.getRandomValues(new Float16Array(6))
+//     }, "Float16Array")
 
-    assert_throws_dom("TypeMismatchError", function() {
-        const len = 65536 / Float16Array.BYTES_PER_ELEMENT + 1;
-        self.crypto.getRandomValues(new Float16Array(len));
-    }, "Float16Array (too long)")
-}, "Float16 arrays");
+//     assert_throws_dom("TypeMismatchError", function() {
+//         const len = 65536 / Float16Array.BYTES_PER_ELEMENT + 1;
+//         self.crypto.getRandomValues(new Float16Array(len));
+//     }, "Float16Array (too long)")
+// }, "Float16 arrays");
 
 test(function() {
     assert_throws_dom("TypeMismatchError", function() {
