diff --git a/WebCryptoAPI/encrypt_decrypt/rsa.js b/WebCryptoAPI/encrypt_decrypt/rsa.js
index 5eae06e47..4be7c37a5 100644
--- a/WebCryptoAPI/encrypt_decrypt/rsa.js
+++ b/WebCryptoAPI/encrypt_decrypt/rsa.js
@@ -1,6 +1,6 @@
 
 function run_test() {
-    var subtle = self.crypto.subtle; // Change to test prefixed implementations
+    var subtle = crypto.subtle; // Change to test prefixed implementations
 
     // When are all these tests really done? When all the promises they use have resolved.
     var all_promises = [];
@@ -291,13 +291,7 @@ function run_test() {
         });
 
         all_promises.push(promise);
-    });
-
-    promise_test(function() {
-        return Promise.all(all_promises)
-            .then(function() {done();})
-            .catch(function() {done();})
-    }, "setup");
+    });    
 
     // A test vector has all needed fields for encryption, EXCEPT that the
     // key field may be null. This function replaces that null with the Correct
